src/h5web/vis-packs/core/complex/MappedComplexVis.tsx

```diff
@@ -62,18 +66,18 @@ function MappedPhaseMapVis(props: Props) {
     phaseDomain = DEFAULT_DOMAIN,
     amplitudeArray,
     amplitudeDomain = DEFAULT_DOMAIN,
-  } = usePhaseAmplitude(mappedArray, colorScaleType);
+  } = usePhaseAmplitude(mappedArray, scaleType);
 
-  const visPhaseDomain = useVisDomain(customDomain, phaseDomain);
-  const [safePhaseDomain] = useSafeDomain(
-    visPhaseDomain,
-    phaseDomain,
-    scaleType
-  );
+  const dataArray =
+    visType !== ComplexVisType.Amplitude ? phaseArray : amplitudeArray;
+  const dataDomain =
+    visType !== ComplexVisType.Amplitude ? phaseDomain : amplitudeDomain;
 
+  const visDomain = useVisDomain(customDomain, dataDomain);
+  const [safeDomain] = useSafeDomain(visDomain, dataDomain, scaleType);
   useEffect(() => {
-    setDataDomain(phaseDomain);
-  }, [phaseDomain, setDataDomain]);
+    setDataDomain(dataDomain);
+  }, [dataDomain, setDataDomain, visType, scaleType]);
```

Comment:
`dataDomain` needs to be reset when changing the `visType` or the `scaleType` so I added these deps to the `useEffect`.